### DISCUSSION TITLE
SizeBasedTriggeringPolicy counter not resetting after TimeBasedTriggeringPolicy rollover

### DISCUSSION THREAD

**Discussion by RollsCorey** [apache/logging-log4j2#1842] [Q&A]
Hello.  Using log4j 2.20.0.  My configuration file specifies a SizeBasedTriggeringPolicy and a TimeBasedTriggeringPolicy.  My understanding is that when the log rolls due to the TimeBasedTriggeringPolicy, the counter associated with the SizeBasedTriggeringPolicy should reset to 1, but this isn't happening.  It just keeps incrementing from where it left off.  Here is my config file.  Any ideas?  Thanks in advance.

```
<Configuration status="warn">
    <Properties>
        <Property name="basePath">Y:/logs</Property>
    </Properties>

    <Appenders>
        <RollingFile name="fileLogger"
                     filePattern="${basePath}/log_%d{yyyyMMdd}_%i.txt">
            <PatternLayout>
                <pattern>%d{MM/dd/yy HH:mm:ss.SSS} %msg%n
                </pattern>
            </PatternLayout>
            <Policies>
                <SizeBasedTriggeringPolicy size="10 MB"/>
                <TimeBasedTriggeringPolicy />
            </Policies>
            <DirectWriteRolloverStrategy maxFiles="10000"/>
        </RollingFile>
    </Appenders>
    <Loggers>
        <Root level="info" additivity="false">
            <appender-ref ref="fileLogger" />
        </Root>
    </Loggers>
</Configuration>
```

**Comment by ppkarwasz** [comment]
@rgoers, can you take a look at this?

**Comment by RollsCorey** [comment]
Hello.  Was wondering if there is some update on this?  The behavior seems to conflict with log4j documentation.  Thank you.

**Reply by rgoers** [reply]
@ppkarwasz Sorry Piotr. I am behind in following these. Yes, I will take a look

**Reply by RollsCorey** [reply]
Really sorry to ask again, but ... any update?  Thank you.

**Comment by hungnguyenmanh82** [comment]
I think It is not a bug. If app still run then %i will increase. If app restarts %i = 1. When I see these fileName with %i=1 , I can known when app start/stop easily.  but you should change config as following (-1 means max of %i = Max of Integer ):
`<DirectWriteRolloverStrategy maxFiles="-1"/>`

**Comment by ppkarwasz** [comment]
It is a bug. I created #2297 to fix it.

**Comment by RollsCorey** [comment]
Thank you!

From: Piotr P. Karwasz ***@***.***>
Sent: Thursday, August 22, 2024 10:02 AM
To: apache/logging-log4j2 ***@***.***>
Cc: Clark, Dave ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [apache/logging-log4j2] SizeBasedTriggeringPolicy counter not resetting after TimeBasedTriggeringPolicy rollover (Discussion #1842)


It is a bug. I created #2297<https://urldefense.com/v3/__https:/github.com/apache/logging-log4j2/issues/2297__;!!NUnUjx3wvH5xgA!eXZQxiymWF9PgDiQMsy37DqD3YvUw6pHDEBjPwoCfRc40iNx105mBFvawgIzeCAgprTHAPE3iLFFxCXpgSWHQi2A-0g$> to fix it.
